lib/cortex/snippets/client/helper.rb

```diff
@@ -13,64 +15,12 @@ def snippet(options = {}, &block)
           end
         end
 
-        def seo_title
-          webpage[:seo_title]
-        end
-
-        def seo_description
-          webpage[:seo_description]
-        end
-
-        def seo_keywords
-          webpage[:seo_keyword_list]
-        end
-
-        def seo_robots
-          robot_information = []
-          index_options = [:noindex, :nofollow, :noodp, :nosnippet, :noarchive, :noimageindex]
-
-          index_options.each do |index_option|
-            robot_information << index_option if webpage[index_option]
+        def current_webpage(cortex_client)
+          if defined?(Rails)
+            Cortex::Snippets::Client::Webpage.new(request, cortex_client)
```

Comment:
Or, actually, my original suggestion still holds - caching should happen here for two reasons:
1) A view helper already implies a dependency on Rails, so the caching logic won't have to check for such a thing.
2) We can cache the expensive instantiation of the `Webpage` class for every `snippet` on a page - caching at the `Webpage` class would lose this benefit